### PR TITLE
beautifulsoup4 4.12.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Jira ticket:** [PKG-1093](https://anaconda.atlassian.net/browse/PKG-1093) (beautifulsoup4-4.12.0)

**The upstream data:**

Changelog: https://git.launchpad.net/beautifulsoup/tree/CHANGELOG
License: https://git.launchpad.net/beautifulsoup/tree/LICENSE
Requirements: https://git.launchpad.net/beautifulsoup/tree/setup.py

**_Actions:_**

1. Rename a package
2. Move tests to the subpackage
3. Add run_constrained (optional but useful dependencies)
4. Fix license_file
5. Remove doc_source_url

**_Notes:_**
 * I've tested it with downstream packages: conda-build, grayskull, moto

**Additional info:**

<details>

**Package's statistics**

<details>

 * Priority B | effort: easy | Category: miniconda | subcategory: core | pkg_type: python
 * Outdated platforms: 'linux-64', 'linux-aarch64', 'linux-s390x', 'linux-ppc64le', 'osx-64', 'osx-arm64', 'win-64'
 * Latest version: 4.12.0
 * Release on PyPi:
    * version: 4.12.0
    * date: 2023-03-20T11:46:31
* [PyPi history](https://pypi.org/project/beautifulsoup4/#history)
 * Popularity: 
    * 3 months downloads: 1526014.0
    * All time:  13061410 downloads -  beautifulsoup4  4.12.0  Python library designed for screen-scraping  

</details>

**Other checks:**

<details>

1. - [x] Check the pinnings
6. - [x] Verify that the 'build_number' is correct
7. - [x] has 'setuptools'
8. - [x] has 'wheel'
9. - [x] 'pip' in test
10. - [x] Verify the test section
11. - [x] Verify if the package is 'architecture specific'
12. - [x] Verify that private modules are not mentioned in the recipe For example: (_private_module)

14. - [x] license_family MIT is present
16. - [x] License: MIT
    - [x] License is 'spdx' compliant
 * Check if the license identifier has correct name from the SPDX License List

| Identifier                       |
|:---------------------------------|
| HPND-sell-variant-MIT-disclaimer |
| MIT                              |
| MIT-0                            |
| MIT-advertising                  |
| MIT-CMU                          |
| MIT-enna                         |
| MIT-feh                          |
| MIT-Modern-Variant               |
| MIT-open-group                   |
| MIT-Wu                           |
| MITNFA                           |
</details>

**Check dependency issues:**

<details>
../aggregate/beautifulsoup4-feedstock/recipe/meta.yaml
Dependencies: ['pip', 'python', 'setuptools', 'beautifulsoup4', 'wheel', 'soupsieve >=1.2']

noarch:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

linux-64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

linux-ppc64le:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

linux-s390x:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

osx-64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

osx-arm64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

win-64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found
- py3.11:  No issues found

</details>
**Other:**

<details>
Checking 'beautifulsoup4' recipe...
~/OneDrive - SoftServe, Inc/src/aggregate/beautifulsoup4-feedstock/recipe ~/OneDrive - SoftServe, Inc/src/finder
Undefined jinja expressions:
    - pin_subpackage
[33m >>> Multi-output recipe -- COMPLICATED [0m
--- OUTPUT: beautifulsoup4
--- Build Requirements ---
>> Empty Host Section <<
--- Host Requirements ---
pip: main
setuptools: main
wheel: main
--- Run Requirements ---
soupsieve >=1.2: main

--- OUTPUT: bs4
--- Build Requirements ---
>> Empty Host Section <<
--- Host Requirements ---
>> Empty Host Section <<
--- Run Requirements ---
[91m  -- jinja directive (compiler?) -- [0m

--- Build Requirements ---
>> Empty Host Section <<
--- Host Requirements ---
>> Empty Host Section <<
--- Run Requirements ---
[91m  -- jinja directive (compiler?) -- [0m
~/OneDrive - SoftServe, Inc/src/finder

</details>

<details>

https://repo.anaconda.com/pkgs/main/noarch
https://repo.anaconda.com/pkgs/main/linux-64
https://repo.anaconda.com/pkgs/main/linux-ppc64le
https://repo.anaconda.com/pkgs/main/linux-aarch64/
https://repo.anaconda.com/pkgs/main/linux-s390x/
https://repo.anaconda.com/pkgs/main/osx-64
https://repo.anaconda.com/pkgs/main/osx-arm64/
https://repo.anaconda.com/pkgs/main/win-64

</details>

<details>

                  name  version                                   spec
0    _anaconda_depends  2022.10                         beautifulsoup4
1    _anaconda_depends  2022.05                         beautifulsoup4
2    _anaconda_depends  2021.11                         beautifulsoup4
3    _anaconda_depends  2021.05                         beautifulsoup4
4    _anaconda_depends  2021.04                         beautifulsoup4
5    _anaconda_depends  2020.07                         beautifulsoup4
6    _anaconda_depends  2020.02                         beautifulsoup4
7    _anaconda_depends  2019.10                         beautifulsoup4
8    _anaconda_depends  2019.07                         beautifulsoup4
9    _anaconda_depends  2019.03                         beautifulsoup4
10   _anaconda_depends  2018.12                         beautifulsoup4
11   _anaconda_depends    5.3.1                         beautifulsoup4
12   _anaconda_depends    5.3.0                         beautifulsoup4
13   _anaconda_depends    5.2.0                         beautifulsoup4
14   _anaconda_depends    5.1.0                         beautifulsoup4
15   _anaconda_depends    5.0.1                         beautifulsoup4
16   _anaconda_depends  5.0.0.1                         beautifulsoup4
17   _anaconda_depends    5.0.0                         beautifulsoup4
18            anaconda  2022.10  beautifulsoup4 4.11.1 py310h06a4308_0
19            anaconda  2022.05   beautifulsoup4 4.11.1 py39ha847dfd_0
20            anaconda  2021.11     beautifulsoup4 4.10.0 pyh06a4308_0
21            anaconda  2021.05      beautifulsoup4 4.9.3 pyha847dfd_0
22            anaconda  2021.04      beautifulsoup4 4.9.3 pyha847dfd_0
23            anaconda  2020.11      beautifulsoup4 4.9.3 pyhb0f4dca_0
24            anaconda  2020.07            beautifulsoup4 4.9.1 py37_0
25            anaconda  2020.02            beautifulsoup4 4.8.2 py36_0
26            anaconda  2019.10            beautifulsoup4 4.8.0 py37_0
27            anaconda  2019.07            beautifulsoup4 4.7.1 py27_1
28            anaconda  2019.03            beautifulsoup4 4.7.1 py27_1
29            anaconda  2018.12            beautifulsoup4 4.6.3 py27_0
30            anaconda    5.3.1            beautifulsoup4 4.6.3 py27_0
31            anaconda    5.3.0            beautifulsoup4 4.6.3 py37_0
32            anaconda    5.2.0    beautifulsoup4 4.6.0 py36hd4cc5e8_1
33            anaconda    5.1.0    beautifulsoup4 4.6.0 py36hd4cc5e8_1
34            anaconda    5.0.1    beautifulsoup4 4.6.0 py36hd4cc5e8_1
35            anaconda  5.0.0.1    beautifulsoup4 4.6.0 py36h49b8c8c_1
36            anaconda    5.0.0    beautifulsoup4 4.6.0 py36hd4cc5e8_1
37                 bs4   4.11.1    beautifulsoup4 >=4.11.1,<4.11.2.0a0
38                 bs4   4.10.0    beautifulsoup4 >=4.10.0,<4.10.1.0a0
39                 bs4    4.9.3      beautifulsoup4 >=4.9.3,<4.9.4.0a0
40                 bs4    4.9.1      beautifulsoup4 >=4.9.1,<4.9.2.0a0
41                 bs4    4.9.0      beautifulsoup4 >=4.9.0,<4.9.1.0a0
42         conda-build   3.23.3                         beautifulsoup4
43         conda-build   3.23.2                         beautifulsoup4
44         conda-build   3.23.1                         beautifulsoup4
45         conda-build   3.23.0                         beautifulsoup4
46         conda-build   3.22.0                         beautifulsoup4
47         conda-build   3.21.9                         beautifulsoup4
48         conda-build   3.21.8                         beautifulsoup4
49         conda-build   3.21.7                         beautifulsoup4
50         conda-build   3.21.6                         beautifulsoup4
51         conda-build   3.21.5                         beautifulsoup4
52         conda-build   3.21.4                         beautifulsoup4
53         conda-build   3.21.3                         beautifulsoup4
54         conda-build   3.21.1                         beautifulsoup4
55         conda-build   3.21.0                         beautifulsoup4
56         conda-build   3.20.5                         beautifulsoup4
57         conda-build   3.20.4                         beautifulsoup4
58         conda-build   3.20.3                         beautifulsoup4
59         conda-build   3.20.2                         beautifulsoup4
60         conda-build   3.20.1                         beautifulsoup4
61         conda-build   3.20.0                         beautifulsoup4
62         conda-build  3.18.11                         beautifulsoup4
63         conda-build  3.18.10                         beautifulsoup4
64         conda-build   3.18.9                         beautifulsoup4
65         conda-build   3.18.8                         beautifulsoup4
66         conda-build   3.18.7                         beautifulsoup4
67         conda-build   3.18.5                         beautifulsoup4
68         conda-build   3.17.8                         beautifulsoup4
69         conda-build   3.17.7                         beautifulsoup4
70         conda-build   3.17.6                         beautifulsoup4
71         conda-build   3.17.5                         beautifulsoup4
72         conda-build   3.17.4                         beautifulsoup4
73         conda-build   3.17.3                         beautifulsoup4
74         conda-build   3.17.1                         beautifulsoup4
75         conda-build   3.17.0                         beautifulsoup4
76         conda-build   3.16.3                         beautifulsoup4
77         conda-build   3.16.2                         beautifulsoup4
78         conda-build   3.16.1                         beautifulsoup4
79         conda-build   3.16.0                         beautifulsoup4
80         conda-build   3.15.1                         beautifulsoup4
81         conda-build   3.15.0                         beautifulsoup4
82         conda-build   3.14.4                         beautifulsoup4
83         conda-build   3.14.2                         beautifulsoup4
84         conda-build   3.14.1                         beautifulsoup4
85         conda-build   3.14.0                         beautifulsoup4
86         conda-build   3.13.0                         beautifulsoup4
87         conda-build   3.12.1                         beautifulsoup4
88         conda-build   3.12.0                         beautifulsoup4
89         conda-build   3.11.0                         beautifulsoup4
90         conda-build   3.10.9                         beautifulsoup4
91         conda-build   3.10.8                         beautifulsoup4
92         conda-build   3.10.7                         beautifulsoup4
93         conda-build   3.10.6                         beautifulsoup4
94         conda-build   3.10.5                         beautifulsoup4
95         conda-build   3.10.4                         beautifulsoup4
96         conda-build   3.10.3                         beautifulsoup4
97         conda-build   3.10.2                         beautifulsoup4
98         conda-build   3.10.1                         beautifulsoup4
99         conda-build   3.10.0                         beautifulsoup4
100        conda-build    3.9.2                         beautifulsoup4
101        conda-build    3.9.1                         beautifulsoup4
102        conda-build    3.8.1                         beautifulsoup4
103        conda-build    3.8.0                         beautifulsoup4
104        conda-build    3.7.2                         beautifulsoup4
105        conda-build    3.7.1                         beautifulsoup4
106        conda-build    3.7.0                         beautifulsoup4
107        conda-build    3.6.0                         beautifulsoup4
108        conda-build    3.5.1                         beautifulsoup4
109        conda-build    3.5.0                         beautifulsoup4
110        conda-build    3.4.2                         beautifulsoup4
111        conda-build    3.4.1                         beautifulsoup4
112        conda-build    3.4.0                         beautifulsoup4
113        conda-build    3.3.0                         beautifulsoup4
114        conda-build    3.2.2                         beautifulsoup4
115        conda-build    3.2.1                         beautifulsoup4
116        conda-build    3.2.0                         beautifulsoup4
117        conda-build    3.1.6                         beautifulsoup4
118        conda-build    3.1.5                         beautifulsoup4
119        conda-build    3.1.4                         beautifulsoup4
120        conda-build    3.1.3                         beautifulsoup4
121        conda-build    3.1.2                         beautifulsoup4
122        conda-build    3.1.1                         beautifulsoup4
123        conda-build    3.1.0                         beautifulsoup4
124        conda-build   3.0.31                         beautifulsoup4
125        conda-build   3.0.30                         beautifulsoup4
126        conda-build   3.0.29                         beautifulsoup4
127        conda-build   3.0.28                         beautifulsoup4
128        conda-build   3.0.27                         beautifulsoup4
129        conda-build   3.0.26                         beautifulsoup4
130        conda-build   3.0.25                         beautifulsoup4
131        conda-build   3.0.23                         beautifulsoup4
132        conda-build   3.0.22                         beautifulsoup4
133          grayskull    2.2.2                         beautifulsoup4
134           libpysal    4.5.1                         beautifulsoup4
135           libpysal    4.2.2                         beautifulsoup4
136               moto    2.2.9                         beautifulsoup4
137               moto    2.1.0                         beautifulsoup4
138               moto   2.0.11                         beautifulsoup4
139               moto    2.0.8                         beautifulsoup4
140               moto    2.0.7                         beautifulsoup4
141               moto    2.0.6                         beautifulsoup4
142               moto    2.0.5                         beautifulsoup4
143               moto    2.0.4                         beautifulsoup4
144               moto    2.0.1                         beautifulsoup4
145          nbconvert    6.5.4                         beautifulsoup4
146          nbconvert    6.4.4                         beautifulsoup4
147            nbsmoke    0.6.0                         beautifulsoup4
148            nbsmoke    0.5.0                         beautifulsoup4
149            nbsmoke    0.4.1                         beautifulsoup4
150            nbsmoke    0.3.0                         beautifulsoup4
151            nbsmoke    0.2.8                         beautifulsoup4
152            nbsmoke    0.2.7                         beautifulsoup4
153          soupsieve    2.2.1                         beautifulsoup4
154          soupsieve      2.2                         beautifulsoup4
155          soupsieve      2.1                         beautifulsoup4
156            webtest   2.0.33                         beautifulsoup4
157            webtest   2.0.32                         beautifulsoup4
158            webtest   2.0.30                         beautifulsoup4
159            webtest   2.0.29                         beautifulsoup4

</details>

expected token 'end of print statement', got 'x'
None

</details>

**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/beautifulsoup4-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/beautifulsoup4-feedstock/tree/4.12.0)
* [conda-forge recipe](https://github.com/conda-forge/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/beautifulsoup4)
* [Diff between upstream and feature branch](https://github.com/conda-forge/beautifulsoup4-feedstock/compare/main...AnacondaRecipes:4.12.0)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/beautifulsoup4-feedstock/compare/master...AnacondaRecipes:4.12.0)
* [The last merged PRs](https://github.com/AnacondaRecipes/beautifulsoup4-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 4.12.0 git@github.com:AnacondaRecipes/beautifulsoup4-feedstock.git
```

[PKG-1093]: https://anaconda.atlassian.net/browse/PKG-1093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,16 @@
-{% set version = "4.11.1" %}
+{% set version = "4.12.0" %}
 
 package:
-  name: beautifulsoup4
+  name: beautifulsoup4-split
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:4]: can you explain why we change the name ?

Comment (skupr-anaconda) [reply recipe/meta.yaml:4]: The linter says: `ERROR: /tmp/abs_071e1cwsmh/clone/recipe/meta.yaml:16: outputs_not_unique: output "beautifulsoup4": Output name is not unique`

Comment (skupr-anaconda) [reply recipe/meta.yaml:4]: https://github.com/anaconda-distribution/perseverance-skills/blob/main/sections/Reference/Recipe_standards.md#the-package-section

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:4]: find ./ -name meta.yaml -exec grep "name: beautifulsoup4" {} +
./beautifulsoup4-feedstock/recipe/meta.yaml:  name: beautifulsoup4
./beautifulsoup4-feedstock/recipe/meta.yaml:  - name: beautifulsoup4

the second found is from outputs

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,41 +25,44 @@ outputs:
...
+          - bs4
+          - bs4.builder
+        requires:
+          - pip
+        commands:
+          - pip check
 
   - name: bs4
     build:
       noarch: generic
```
[the remark this answers is not in the document]
Comment (danpetry) [reply recipe/meta.yaml:42]: `noarch: python` isn't allowed but `noarch: generic` is ok: https://github.com/anaconda-distribution/perseverance-skills/blob/main/sections/Team_processes/Policy_changelog.md#no-more-noarch-python-builds

File: recipe/meta.yaml [hunk 3]
```diff
@@ -26,41 +25,44 @@ outputs:
       run:
         - python
         - soupsieve >=1.2
```
Comment (danpetry) [inline recipe/meta.yaml:27]: Upstream says >1.2 not >=1.2

File: recipe/meta.yaml [hunk 4]
```diff
@@ -26,41 +25,44 @@ outputs:
...
+      run_constrained:
+        - html5lib
+        - lxml
```
Comment (danpetry) [inline recipe/meta.yaml:30]: AFAIU there's no point in specifying packages in `run_constrained` without pinning: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html?highlight=run_constrained#run-constrained

Comment (danpetry) [reply recipe/meta.yaml:30]: I think you would just judge whether you want these packages or not (should have a good reason to include them if so) and if you're including them, just put them in the run section

File: recipe/meta.yaml [hunk 5]
```diff
@@ -26,41 +25,44 @@ outputs:
...
       noarch: generic
     requirements:
+      host:
+        - python
+        - pip
+        - setuptools
+        - wheel
       run:
+        - python
         - {{ pin_subpackage('beautifulsoup4', max_pin="x.x.x") }}
```
Comment (danpetry) [inline recipe/meta.yaml:51]: As mentioned on slack, I'm not sure what's the point in the `bs4` output?

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:27]: Fixed

File: recipe/meta.yaml (hunk 4 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:30]: I'll remove it. The size of the package matters

File: recipe/meta.yaml (hunk 5 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:51]: It's a common name for `import bs4`

Comment (danpetry) [reply recipe/meta.yaml:48]: Right, so conda users will expect to be able to `conda install bs4`?

Comment (skupr-anaconda) [reply recipe/meta.yaml:48]: Yes

**Review by cbouss — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**